doc/nimc.rst

```diff
@@ -326,20 +326,66 @@ They are:
 DLL generation
 ==============
 
-Nim supports the generation of DLLs. However, there must be only one
-instance of the GC per process/address space. This instance is contained in
-``nimrtl.dll``. This means that every generated Nim DLL depends
-on ``nimrtl.dll``. To generate the "nimrtl.dll" file, use the command::
+Nim supports the generation of dynamic libraries. However, there must be only
+one instance of the GC per process/address space, meaning that if multiple Nim
+DLLs are used by the same binary, the DLLs must link to the ``nimrtl`` dynamic 
+library.
+
+**Note for UNIX**: If any Nim DLL (including ``libnimrtl.so`` and
+``libnimchr.so``) is to be directly included with your application in the same
+folder as the app binary, the app's ``rpath`` variable should be set to
+``$ORIGIN``. The reason for this is because the dynamic library linker for
+UNIX-like systems *does not* search for dynamic libraries in the same folder
+(unlike Windows), unless the ``rpath`` variable is set otherwise in the binary
+to search for in the app directory.
+
+For an app binary built in Nim, an easy way to set the ``rpath`` variable
+correctly is by including the following line in your source code:
 
-  nim c -d:release lib/nimrtl.nim
+.. code-block:: Nim
+  when defined(unix): {.passL: "-Wl,-rpath='$ORIGIN'"}
+
+``nimrtl``
+----------
+
+The ``nimrtl`` library contains both Nim's memory manager as well as the full
+standard library. To generate the ``nimrtl`` library, first, create a file named
+``build_nimrtl.nim``, and fill it with the following:
+
+.. code-block:: Nim
+  include nimrtl
```

Comment:
Huh? No, 'nimrtl' is a "main Nim" file, not an include file.

Reply:
I don't think this is ideal too, but the problem is this is the easiest way I can think of to generate nimrtl/hcr from anywhere without having to change the compiler. The compiler won't accept `lib/nimrtl.nim` if we're not in the stdlib directory, and it's annoying for the user to have to find it (especially if `choosenim` is being used)

Reply:
Hmm ok.

Reply:
I ended up just adding a new compilation option anyway